src/protocol/basics/sum_of_product/malicious.rs

```diff
@@ -130,9 +130,11 @@ mod test {
 
         let res = world
             .malicious((av, bv), |ctx, (a_share, b_share)| async move {
-                let a_refs = a_share.iter().collect::<Vec<_>>();
-                let b_refs = b_share.iter().collect::<Vec<_>>();
-                ctx.sum_of_products(RecordId::from(0), a_refs.as_slice(), b_refs.as_slice())
+                let mut pairs = Vec::with_capacity(MULTI_BIT_LEN);
```

Comment:
MULTI_BIT_LEN does not belong here. This file is just for the abstract dot-product computation. Can you rename it something generic like VECTOR_LENGTH?

Reply:
Changed to BATCHSIZE.